### PR TITLE
OpenVDB Node

### PR DESCRIPTION
**Pull request by Theverat**

### CODE REVIEW

File: handlers/frame_change_pre.py [hunk 1]
```diff
@@ -1,31 +1,49 @@
...
         if not mat.luxcore.node_tree:
             continue
 
+        if utils_node.find_nodes(mat.luxcore.node_tree, "LuxCoreNodeTexOpenVDB") or \
+                utils_node.find_nodes(mat.luxcore.node_tree, "LuxCoreNodeTexSmoke"):
+            found_smoke_sequence = True
+            for n in mat.luxcore.node_tree.nodes:
+                if n.name.split('.')[0] == "Material Output" and n.active:
+                    # Force a viewport update
+                    mat.luxcore.node_tree.links.new(n.inputs["Material"], n.inputs["Material"].links[0].from_socket)
```
Comment (Theverat) [inline handlers/frame_change_pre.py]: a) I don't like single-letter names, they make the code hard to read, please write out "node" in this case.

b) You should never make any decision based on `node.name`, it is a completely arbitrary name that can even be changed by the user in the UI. 
Use `node.bl_idname == "LuxCoreNodeMatOutput"` instead.

c) I do not understand why you search for an output node at all?
To force a viewport update, just use the same code as I did below: `mat.diffuse_color = mat.diffuse_color`

d) `break` from the loop after you forced the material update to save performance

File: nodes/textures/openVDB.py [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+                for link in e.links:
+                    links.append(link.to_socket)
+                old_sockets[e.name] = links.copy()
+
+            self.outputs.clear()
+            names = pyluxcore.GetOpenVDBGridNames(bpy.path.abspath(self.file_path))
+            self.has_high_resolution = False
+            self.use_high_resolution = False
+
+            for n in names:
```
Comment (Theverat) [inline nodes/textures/openVDB.py]: Please use `name` instead of single-letter `n`.

File: handlers/frame_change_pre.py (hunk 1 above)
Comment (neo2068) [reply handlers/frame_change_pre.py]: > c) I do not understand why you search for an output node at all?
> To force a viewport update, just use the same code as I did below: mat.diffuse_color = mat.diffuse_color

The openVDB node is used with volumes and null material which don't have a diffuse_color.

Comment (Theverat) [reply handlers/frame_change_pre.py]: Ok, but in the code above, it looks to me like you are editing the material's node tree to force an update, not some volume node tree. I have to admit I don't understand why directly editing the material wouldn't lead to the same result?

**Pull request merged**